versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/MongoDBConditionAliasVisitor.java

```diff
@@ -0,0 +1,113 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl.condition;
+
+import java.util.List;
+
+import com.dremio.nessie.versioned.store.Entity;
+import com.google.common.collect.ImmutableList;
+
+/*
+ * This uses the Visitor design pattern to retrieve object attributes.
+ */
+public class MongoDBConditionAliasVisitor implements ConditionAliasVisitor {
+  /**
+   * The callback method in the visitor design pattern.
+   * @param conditionExpression The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ConditionExpression.
+   */
+  @Override
+  public ConditionExpression visit(ConditionExpression conditionExpression, AliasCollector collector) {
+    return ImmutableConditionExpression.builder()
+      .functions(conditionExpression.getFunctions().stream()
+        .map(f -> f.acceptExpressionFunction(this, collector)).collect(ImmutableList.toImmutableList()))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionFunction The object to be aliased.
+   * @param arguments constituent arguments of the object to be aliased.
+   * @param name the name of the object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ExpressionFunction.
+   */
+  @Override
+  public ExpressionFunction visit(ExpressionFunction expressionFunction, List<Value> arguments,
+                                  ExpressionFunction.FunctionName name, AliasCollector collector) {
+    return new ExpressionFunction(name, arguments.stream().map(v -> getArgumentValue(v, collector))
+      .collect(ImmutableList.toImmutableList()));
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionpath The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  @Override
+  public ExpressionPath visit(ExpressionPath expressionpath, AliasCollector collector) {
+    return ImmutableExpressionPath.builder()
+      .root((ExpressionPath.NameSegment) expressionpath.getRoot().accept(this, collector))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param nameSegment The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  @Override
+  public ExpressionPath.NameSegment visit(ExpressionPath.NameSegment nameSegment, AliasCollector collector) {
+    return ImmutableNameSegment.builder()
```

Comment:
Where can this be visited from, the parent ExpressionPath? Wondering if this should be a separate method if so.

Reply:
Yes from ExpressionPath. If I expose those attributes I need in ExpressionPath then I think this method will simplify but it is mirroring the behaviour of aliasing in ExpressionPath.NameSegment. Also at this stage I know PathSegment will be used for UpdateExpression, which NameSegment extends, so I would rather leave this in until that is written. It could be this method moves up to that parent class.